internal/services/vmware/vmware_private_cloud_netapp_file_volume_attachment_resource_test.go

```diff
@@ -0,0 +1,231 @@
+package vmware_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/vmware/2021-12-01/datastores"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type VmwareNetappFileVolumeAttachmentResource struct{}
+
+func TestAccVmwarePrivateCloudNetappFileVolumeAttachment_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_vmware_netapp_volume_attachment", "test")
+	r := VmwareNetappFileVolumeAttachmentResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r)),
+		},
+		data.ImportStep(),
+	})
+}
+
+func (r VmwareNetappFileVolumeAttachmentResource) Exists(ctx context.Context, client *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := datastores.ParseDataStoreID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := client.Vmware.DataStoreClient.Get(ctx, *id)
+	if err != nil {
+		if response.WasNotFound(resp.HttpResponse) {
+			return utils.Bool(false), nil
+		}
+		return nil, fmt.Errorf("retrieving %s: %+v", *id, err)
+	}
+	return utils.Bool(true), nil
+}
+
+func (r VmwareNetappFileVolumeAttachmentResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestrg-vmware-nat-%d"
+  location = "centralus"
+}
+
+%s
+
+%s
+resource "azurerm_vmware_netapp_volume_attachment" "test" {
+  name              = "acctest-vmwareattachment-%d"
+  netapp_volume_id  = azurerm_netapp_volume.test.id
+  vmware_cluster_id = "${azurerm_vmware_private_cloud.test.id}/clusters/Cluster-1"
+
+  depends_on = [azurerm_virtual_network_gateway_connection.test]
+}`, data.RandomInteger, r.templatePrivateCloud(data), r.templateNetappFile(data), data.RandomInteger)
+}
+
+func (r VmwareNetappFileVolumeAttachmentResource) templatePrivateCloud(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {
+    resource_group {
+      prevent_deletion_if_contains_resources = false
+    }
+  }
+  # In Vmware acctest, please disable correlation request id, else the continuous operations like update or delete will not be triggered
+  # issue https://github.com/Azure/azure-rest-api-specs/issues/14086 
```

Comment:
waiting on https://github.com/Azure/azure-rest-api-specs/issues/14086#issuecomment-1298625805

Reply:
To be safe, I'd leave the workaround in place in November.

Reply:
@cataggar - this is only for the tests and won't affect end users, we should be removing it now so the tests represent what a user experiences